tests/observability/metrics/conftest.py

```diff
@@ -228,22 +227,41 @@ def updated_resource_multiple_times_with_invalid_label(
     updated_value = 0
     for index in range(count):
         increasing_value += 1
-        resource_editor = ResourceEditor(
-            patches={
-                resource: {
-                    "metadata": {
-                        "labels": {VERSION_LABEL_KEY: None},
-                    },
+        max_retries = 3
```

Comment:
the logic is too complex, please split to smaller functions and use `@retry` when possible or timeout sampler if not